src/solver.jl

```diff
@@ -380,21 +380,29 @@ function MOI.add_constraint(model::Optimizer, f::MOI.SingleVariable, set::SCALAR
     return MOI.ConstraintIndex{typeof(f), typeof(set)}(vi.value)
 end
 
+MOI.supports_constraint(::Optimizer, ::Type{MOI.SingleVariable}, ::Type{MOI.Integer}) = true
+
 function MOI.add_constraint(model::Optimizer, f::MOI.SingleVariable, set::MOI.Integer)
     model.var_type_orig[f.variable.index] = :Int
 end
+MOI.supports_constraint(::Optimizer, ::Type{MOI.SingleVariable}, ::Type{MOI.ZeroOne}) = true
 
 function MOI.add_constraint(model::Optimizer, f::MOI.SingleVariable, set::MOI.ZeroOne)
     model.var_type_orig[f.variable.index] = :Bin
 end
 
 
-function MOI.add_constraint(model::Optimizer, f::MOI.SingleVariable, set::SCALAR_SET)
+function MOI.add_constraint(model::Optimizer, f::MOI.ScalarAffineFunction, set::SCALAR_SET)
+    model.num_constr_orig += 1
+
 
+    return MOI.ConstraintIndex{typeof(f), typeof(set)}(model.num_const_orig)
+end
```

Comment:
I do not really know where to put this constraint. 
In the MathProgBase version it is stored in `model.d_orig.A` I couldn't find the right hand side though.